packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -393,6 +393,8 @@ class BottomSheet extends Component {
 			children,
 			withHeaderSeparator = false,
 			hasNavigation,
+			// eslint-disable-next-line no-unused-vars
+			onDismiss,
```

Comment:
I noticed that if the `onDismiss` prop is passed to the bottom sheet component, the internal function `onDismiss` ([code reference](https://github.com/WordPress/gutenberg/blob/fa3bfd5e960fb6d9901edc05530a27454ecb912a/packages/components/src/mobile/bottom-sheet/index.native.js#L309-L317)) is not executed. This behavior looks wrong because this function is intended to actually call the `onDismiss` prop. To address it I added the prop to this props destructuring list to prevent including it later on when the component is rendered ([code reference](https://github.com/WordPress/gutenberg/blob/fa3bfd5e960fb6d9901edc05530a27454ecb912a/packages/components/src/mobile/bottom-sheet/index.native.js#L513)).

Reply:
Aha, I see what's going on there, leaving the `onDismiss` in the `rest` was ending up overriding the [attempt to set it here](https://github.com/WordPress/gutenberg/blob/fa3bfd5e960fb6d9901edc05530a27454ecb912a/packages/components/src/mobile/bottom-sheet/index.native.js#L499).

Cool trick to fix it here, though I'm worried this is very esay to break and would at least recommend adding a line comment in the "extraction" point here to inform the reader what's going on.

Alternatively, what do you think about a different way to remove the `onDismiss` preset in rest closer to the code we want the removal? Like, instead of [`{ ...rest }`](https://github.com/WordPress/gutenberg/blob/fa3bfd5e960fb6d9901edc05530a27454ecb912a/packages/components/src/mobile/bottom-sheet/index.native.js#L513) wdyt about removing the onDismiss:

```javascript
				{ ...( { onDismiss: _, ...rest } ) => rest } // removing the `onDismiss` as we're actually setting it earlier above.
```

Reply:
> Cool trick to fix it here, though I'm worried this is very easy to break because it's a bit unintuitive and too far away from the codesite it affects. I'd recommend adding a line comment in the "extraction" point here to inform the reader what's going on.

Sure! I'll add a comment to make it clearer 👍 .

> Alternatively, what do you think about a different way to remove the onDismiss preset in rest closer to the code we want the removal? Like, instead of { ...rest } wdyt about removing the onDismiss

Yeah, that would do the trick, I'm wondering if we could also use [`omit`](https://lodash.com/docs/4.17.15#omit) as we use on other places:
```
{ ...omit( rest, 'onDismiss' ) }
```

Reply:
> Yeah, that would do the trick, I'm wondering if we could also use omit as we use on other places:
>
> { ...omit( rest, 'onDismiss' ) }

Ah, right, we might as well use lodash, it's also more easy to read 👍

Reply:
I addressed this issue in [this commit](https://github.com/WordPress/gutenberg/pull/31343/commits/32c4466f62f4cb6f71e93e3256b76a1649d3b1d4) 🎊 .